.murdock

```diff
@@ -236,19 +244,33 @@ test_job() {
     local appdir=$1
     local board=$(echo $2 | cut -f 1 -d':')
     local toolchain=$(echo $2 | cut -f 2 -d':')
-    local flashfile="$3"
 
-    [ ! -f "$flashfile" ] && {
-        echo "$0: _test(): flashfile \"$flashfile\" doesn't exist!"
-        return 1
-    }
+    # interpret any extra arguments as file names.
+    # They will be sent along with the job to the test worker
+    # and stored in the application's binary folder.
+    shift 2
+    local files=""
+    for filename in "$@"; do
+        # check if the file is within $(BINDIR)
+        if startswith "${BINDIR}" "${filename}"; then
+          # get relative (to $(BINDIR) path
+          local relpath="$(realpath --relative-to ${BINDIR} ${filename})"
```

Comment:
* `realpath` is not supported on `OSx` https://github.com/RIOT-OS/RIOT/pull/10568#issuecomment-445221113 so it avoided in the build system. If we can force them to install it I would gladly use it at some other places.
* `BINDIR` is allowed to be locally overwritten and it is not handled.

If even supported to not be in `BINDIR`, I would handle it relative to `RIOTBASE` and only in a subdirectory of `RIOTBASE`.
Relative previous directories handling is the first thing banned in any services to keep sandboxed.

Reply:
>     * `realpath` is not supported on `OSx` [#10568 (comment)](https://github.com/RIOT-OS/RIOT/pull/10568#issuecomment-445221113) so it avoided in the build system. If we can force them to install it I would gladly use it at some other places.

All CI workers are required to provide realpath. This is a closed ecosystem, if we (finally) add OSX builders, we can surely make them also provide that.



>     * `BINDIR` is allowed to be locally overwritten and it is not handled.

Local BINDIR overrides are handled by using BINDIR as variable. Remote (on workers) BINDIR overrides are currently not handled, as that is currently not needed. There's a comment on that.

> Relative previous directories handling is the first thing banned in any services to keep sandboxed.

A check whether one of the files is in RIOTBASE can be easily added.

Reply:
> A check whether ~one of~ the files ~is~ are in RIOTBASE can be easily added.

they now have to be in BINDIR.